draft-ietf-tls-tls13.md

```diff
@@ -3413,52 +3450,73 @@ negotiated, as it provides no more protection than an unsecured connection.
+    TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256       {0xC0,0x2B}    [RFC5289]
+    TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384       {0xC0,0x2C}    [RFC5289]
+    TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256         {0xC0,0x2F}    [RFC5289]
+    TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384         {0xC0,0x30}    [RFC5289]
+    TLS_DHE_RSA_WITH_AES_128_CCM                  {0xC0,0x9E}    [RFC6655]
+    TLS_DHE_RSA_WITH_AES_256_CCM                  {0xC0,0x9F}    [RFC6655]
+    TLS_DHE_RSA_WITH_AES_128_CCM_8                {0xC0,0xA2}    [RFC6655]
+    TLS_DHE_RSA_WITH_AES_256_CCM_8                {0xC0,0xA3}    [RFC6655]
+    TLS_ECDHE_ECDSA_WITH_AES_128_CCM              {0xC0,0xAC}    [RFC7251]
+    TLS_ECDHE_ECDSA_WITH_AES_256_CCM              {0xC0,0xAD}    [RFC7251]
+    TLS_ECDHE_ECDSA_WITH_AES_128_CCM_8            {0xC0,0xAE}    [RFC7251]
+    TLS_ECDHE_ECDSA_WITH_AES_256_CCM_8            {0xC0,0xAF}    [RFC7251]
+    TLS_DHE_RSA_WITH_ARIA_128_GCM_SHA256          {0xC0,0x52}    [RFC6209]
+    TLS_DHE_RSA_WITH_ARIA_256_GCM_SHA384          {0xC0,0x53}    [RFC6209]
+    TLS_DHE_DSS_WITH_ARIA_128_GCM_SHA256          {0xC0,0x56}    [RFC6209]
+    TLS_DHE_DSS_WITH_ARIA_256_GCM_SHA384          {0xC0,0x57}    [RFC6209]
+    TLS_ECDHE_ECDSA_WITH_ARIA_128_GCM_SHA256      {0xC0,0x5C}    [RFC6209]
+    TLS_ECDHE_ECDSA_WITH_ARIA_256_GCM_SHA384      {0xC0,0x5D}    [RFC6209]
+    TLS_ECDHE_RSA_WITH_ARIA_128_GCM_SHA256        {0xC0,0x60}    [RFC6209]
+    TLS_ECDHE_RSA_WITH_ARIA_256_GCM_SHA384        {0xC0,0x61}    [RFC6209]
+    TLS_DHE_RSA_WITH_CAMELLIA_128_GCM_SHA256      {0xC0,0x7C}    [RFC6367]
+    TLS_DHE_RSA_WITH_CAMELLIA_256_GCM_SHA384      {0xC0,0x7D}    [RFC6367]
+    TLS_DHE_DSS_WITH_CAMELLIA_128_GCM_SHA256      {0xC0,0x80}    [RFC6367]
+    TLS_DHE_DSS_WITH_CAMELLIA_256_GCM_SHA384      {0xC0,0x81}    [RFC6367]
+    TLS_ECDHE_ECDSA_WITH_CAMELLIA_128_GCM_SHA256  {0xC0,0x86}    [RFC6367]
+    TLS_ECDHE_ECDSA_WITH_CAMELLIA_256_GCM_SHA384  {0xC0,0x87}    [RFC6367]
+    TLS_ECDHE_RSA_WITH_CAMELLIA_128_GCM_SHA256    {0xC0,0x8A}    [RFC6367]
+    TLS_ECDHE_RSA_WITH_CAMELLIA_256_GCM_SHA384    {0xC0,0x8B}    [RFC6367]
+    TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305          {0xCC,0xA1}  [I-D.ietf-tls-chacha20-poly1305]
+    TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305        {0xCC,0xA2}  [I-D.ietf-tls-chacha20-poly1305]
+    TLS_DHE_RSA_WITH_CHACHA20_POLY1305            {0xCC,0xA3}  [I-D.ietf-tls-chacha20-poly1305]
+
+[[TODO: CHACHA20_POLY1305 cipher suite IDs are TBD. Must update above if
+final versions change.]]
+
+Note: In the case of the CCM mode of AES, two variations exist: "CCM_8" which
+uses an 8-bit authentication tag and "CCM" which uses a 16-bit authentication
+tag. Both use the default hash, SHA-256.
+
+DHE AES GCM & CCM are standards track ([RFC5288] & [RFC6655]),
+as is ChaChaPoly ([I-D.ietf-tls-chacha20-poly1305]).  The ECDHE variants of AES
+and other ciphers above are currently informational, however ECDHE AES GCM
+is widely deployed.
```

Comment:
On Wednesday, July 15, 2015 02:22:12 pm ekr wrote:
> Let's put this in a table instead.

Which would you prefer:
a) add an asterisk next to the spec links to indicate informational
b) split the table into two for standard and informational

I think (a) is better, because that keeps ECDHE AES GCM next to DHE AES GCM. Either works.

Is it possible to promote cipher suite codepoints to standard status straight in this RFC?


Dave